site/docs/hosting/vps.md

```diff
@@ -45,131 +120,486 @@ cd /etc/systemd/system
 
-```sh
-systemctl disable <service_name>
+Після цього виконайте ще одну команду:
+
+```sh:no-line-numbers
+pm2 save
 ```
 
-## PM2 (Node.js only)
+This command will save the list of current applications so that they can be launched after a server reboot.
 
-PM2 is a daemon process manager for Node.js that will help you manage and keep your app online 24/7.
+If you have created a new application and want to save it as well, simply run `pm2 save` again.
 
-### Installing
+## Running the Bot on Webhooks
 
-```sh
-npm install pm2@latest -g
+To run a bot on webhooks, you will need to use a web framework and not call `bot.start()`.
 
-# If using Yarn
-yarn global add pm2
+Here's a sample code to start the bot using `fastify`, which you should add to the bot's main file:
+
+```ts
+import { webhookCallback } from "grammy";
+import { fastify } from "fastify";
+
+const server = fastify();
+
+server.post(`/${bot.token}`, webhookCallback(bot, "fastify"));
+
+server.listen();
 ```
 
-### Managing Apps
+### Domain Rental
 
-#### Start
+To connect a bot running on webhooks to the outside world, you need to purchase a domain.
 
-```sh
-pm2 start --name <app_name> <entry_point>
+Go to the [domain name search page](https://www.hostinger.com/domain-name-search).
+In the text input field, enter a domain name of the form `<name>.<zone>`.
+For example, `example.com`.
+
+If the desired domain is free, click the Add button next to it.
+You will be automatically redirected to the checkout page, where you will also immediately register with Hostinger if you are not already registered.
+Pay for the domain.
+
+#### Domain Pointing to VPS
+
+Before your domain can work with your VPS, you need to point the domain to your server.
+To do this, in the [Hostinger Control Panel](https://hpanel.hostinger.com/), click the "Manage" button next to your domain.
+Next, go to the DNS record management page by clicking on the "DNS / Name Servers" button in the menu on the left.
+
+> First, find out the IP address of your VPS.
+
+In the list of DNS records, find the record of type `A` with the name `@`.
+Edit this record by changing the IP address in the "Points to" field to the IP address of your VPS, and set the TTL to 3600.
+
+Next, find and delete the record of type `CNAME` with the name `www`.
+Instead, create a new record of type `A` with the name `www`, pointing to the IP address of your VPS, and set the TTL to 3600.
+
+> In the event of a difficulty, use the other method described in [knowledge base](https://support.hostinger.com/en/articles/1583227-how-to-point-domain-to-your-vps).
+
+### Setting up a Web Server
+
+For the website to work and the bot to start receiving updates from Telegram, you need to set up a web server.
+We will use [Caddy](https://caddyserver.com/).
+
+Caddy is a powerful open source web server with automatic HTTPS.
+It is written in Go, so it is very fast. :zap:
```

Comment:
That's not how it works, this is simplifying things too much. Let's not spread false information :)

Reply:
Haha, okay. Should I remove this sentence?